src/SAML2/XML/ds/KeyName.php

```diff
@@ -58,22 +51,33 @@ public function getName(): string
      * @param string $name
      * @return void
      */
-    public function setName(string $name): void
+    private function setName(string $name): void
     {
         $this->name = $name;
     }
 
 
+    /**
+     * Convert XML into a KeyName
+     *
+     * @param \DOMElement $xml The XML element we should load
+     * @return self
+     */
+    public static function fromXML(DOMElement $xml): object
+    {
+        return new self($xml->textContent);
+    }
+
+
     /**
      * Convert this KeyName element to XML.
      *
-     * @param \DOMElement $parent The element we should append this KeyName element to.
+     * @param \DOMElement|null $parent The element we should append this KeyName element to.
      * @return \DOMElement
      */
-    public function toXML(DOMElement $parent): DOMElement
+    public function toXML(DOMElement $parent = null): DOMElement
     {
-        Assert::notEmpty($this->name);
-
-        return Utils::addString($parent, XMLSecurityDSig::XMLDSIGNS, 'ds:KeyName', $this->name);
+//        $e = $this->instantiateParentElement($parent);
+        return Utils::addString($parent, self::NS, 'ds:KeyName', $this->name);
```

Comment:
Either we use `instantiateParentElement($parent)` and add the contents manually, or if we still want to use `Utils::addString()` we should change `'ds:KeyName'` to `static::NS_PREFIX.':'.static::class`.

Reply:
I'll fiddle with it some more to see if I can get it to work

Reply:
This was fairly easy actually... Got it!